production/scrapers/michigan_staff.R

```diff
@@ -16,41 +16,60 @@ michigan_staff_date_check <- function(x, date = Sys.Date()){
         error_on_date(date)
 }
 
-michigan_staff_pull <- function(x){
-    mi_html <- xml2::read_html(x)
-    img2 <- mi_html %>%
-        rvest::html_nodes("img") %>%
-        rvest::html_attr("src") %>%
-        .[7] %>%      #src="https://miro.medium.com/max/1210/1*rmlt07ZUHyPh0MkrayUqoA.png"
-        magick::image_read()
+michigan_staff_pull <- function(url){
+    mi_html <- xml2::read_html(url)
 
-    img2
-}
+    data_div <- rvest::html_node(
+        mi_html,
+        xpath = "//h1[contains(text(), 'Total Confirmed Prisoner')]/parent::div")
+
+    staff_image <- rvest::html_nodes(data_div, "img") %>%
+        .[3] %>%
+        rvest::html_attr("src") %>%
+        magick::image_read() %>%
+        magick::image_crop("0x0+0+40")
 
-michigan_staff_restruct <- function(x){
-    ExtractTable(x)
+    return(staff_image)
 }
 
-michigan_staff_extract <- function(x){
-    col_name_mat <- matrix(c(
-        "Location", "0", "Name",
-        "Staff confirmed", "1", "Staff.Confirmed",
-        "Staff deaths", "2" , "Staff.Deaths"
-    ), ncol = 3, nrow = 3, byrow = TRUE)
+michigan_staff_restruct <- function(staff_image){
+    table_body <- magick::image_crop(staff_image, "0x0+0+30")
     
-    colnames(col_name_mat) <- c("check", "raw", "clean")
-    col_name_df <- as_tibble(col_name_mat)
+    staff_data <- ExtractTable(table_body)
     
-    check_names_extractable(x[[1]], col_name_df)
-    
-    extract_out <- rename_extractable(x[[1]], col_name_df) %>%
+    names(staff_data[[1]]) <- c("Name", "Staff.Confirmed", "Staff.Deaths")
+
+    return(staff_data)
+}
+
+michigan_staff_extract <- function(restructured_data){
+    extract_data <- restructured_data[[1]] %>%
         as_tibble() %>%
-        filter(Name != "Location" & !grepl(Name, pattern = "Tota")) %>%
+        filter(!grepl(Name, pattern = "Tota")) %>%
         clean_scraped_df() %>%
         mutate(Staff.Deaths = ifelse(is.na(Staff.Deaths), 0, Staff.Deaths),
-               Name = str_remove(Name, "^Location ")) 
+               Name = str_remove(Name, "^Location "))
+
+    michigan_staff_check_extracted_data(extract_data)
+
+    return(extract_data)
+}
+
+michigan_staff_check_extracted_data <- function(extract_data){
+    if (str_detect(extract_data$Name, "\\d|ocation")) {
```

Comment:
The way you've written these `if `statements means that the first row is the only one checked.  I think we want to be checking all rows, especially for the alpha checks of the number columns. I would change all of these `if()` statements to something like: 

    if( nrow(extract_data %>%
        filter(str_detect(Name, "\\d|ocation"))) > 0) {
        warning("Examine Name column for mis-extracted data;
                numbers or text 'location' found where not expected")
    }

This has the added benefit of avoiding an expected warning message on every run, so when this functions as expected, we won't have a new log file!

Reply:
oooooo thank you!! I'm going to add this as written now!